contracts/UniswapV2Pair.sol

```diff
@@ -224,24 +176,17 @@ contract UniswapV2Pair is NilBase, IUniswapV2Pair {
         );
         uint balance0;
         uint balance1;
-        balance0 = TokenLibrary(tokenLib).balanceOf(token0, address(this));
-        balance1 = TokenLibrary(tokenLib).balanceOf(token1, address(this));
+        balance0 = Nil.currencyBalance(address(this), tokenId0);
+        balance1 = Nil.currencyBalance(address(this), tokenId1);
         {
             // scope for _token{0,1}, avoids stack too deep errors
             address _token0 = token0;
             address _token1 = token1;
             require(to != _token0 && to != _token1, "UniswapV2: INVALID_TO");
-            if (amount0Out > 0) _safeTransfer(_token0, to, amount0Out); // optimistically transfer tokens
-            if (amount1Out > 0) _safeTransfer(_token1, to, amount1Out); // optimistically transfer tokens
-            if (data.length > 0)
-                IUniswapV2Callee(to).uniswapV2Call(
-                    msg.sender,
-                    amount0Out,
-                    amount1Out,
-                    data
-                );
-            balance0 = TokenLibrary(tokenLib).balanceOf(_token0, address(this));
-            balance1 = TokenLibrary(tokenLib).balanceOf(_token1, address(this));
+            if (amount0Out > 0) _safeTransfer(tokenId0, to, amount0Out); // optimistically transfer tokens
```

Comment:
Now it's not optimistic transfer as we remove flash swaps, so there is not way of reverting it

Reply:
removed